From 7b8b5cb05bd870549dee2d159e246ce58e325b7f Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Tue, 6 Nov 2012 00:10:48 +0200
Subject: [PATCH] Avoid duplicating volume icons

This is a workaround for GVFS 1.14.0 which emits volume-added signals
for pre-existing volumes.
---
 src/xfdesktop-file-icon-manager.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/xfdesktop-file-icon-manager.c b/src/xfdesktop-file-icon-manager.c
index 72b6d09..e039cb8 100644
--- a/src/xfdesktop-file-icon-manager.c
+++ b/src/xfdesktop-file-icon-manager.c
@@ -2521,6 +2521,10 @@ xfdesktop_file_icon_manager_volume_added(GVolumeMonitor *monitor,
 {
     XfdesktopFileIconManager *fmanager = XFDESKTOP_FILE_ICON_MANAGER(user_data);
 
+    /* avoid duplicating volume icons -- workaround for GVFS 1.14.0 */
+    if (g_hash_table_lookup(fmanager->priv->removable_icons, volume))
+        return;
+
     xfdesktop_file_icon_manager_add_removable_volume(fmanager, volume);
 }
 
-- 
1.8.0

